connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfigTransformer.java

```diff
@@ -46,7 +52,15 @@ public WorkerConfigTransformer(Worker worker, Map<String, ConfigProvider> config
         if (configs == null) return null;
         ConfigTransformerResult result = configTransformer.transform(configs);
         if (connectorName != null) {
-            scheduleReload(connectorName, result.ttls());
+            String action = configs.get(ConnectorConfig.CONFIG_RELOAD_ACTION_CONFIG);
```

Comment:
shouldn't the default come from ConfigDef?

Reply:
The config transformation happens before a ConfigDef is created.